models/Document/Editable/Areablock.php

```diff
@@ -248,8 +248,11 @@ protected function getEditableHandler(): EditableHandler
 
     public function setDataFromResource(mixed $data): static
     {
-        $this->indices = Tool\Serialize::unserialize($data);
-        if (!is_array($this->indices)) {
+        $unserializedData = Tool\Serialize::unserialize($data);
+
+        if (is_array($unserializedData)) {
+            $this->indices = $unserializedData;
```

Comment:
$data can be an array
```suggestion
        if (is_string($data)) {
            $data = Tool\Serialize::unserialize($data);
        }

        if (is_array($data)) {
            $this->indices = $data;
```

Reply:
Hm.. This wont work either. If $data is the empty string "" the  Tool\Serialize::unserialize($data); will return a string and you will get the type error.

Reply:
Hmm, but than the `if (is_array($data)) {` should be false and should set an empty array, why not?

Reply:
Not sure what you mean and what you will achieve with your change. If $data is a string it will converted to a an empty array with both your change and without it.
(Also overwriting injected parameters, like $data, I think it might be considered a code smell/anti-pattern or at least bad practice.)

Reply:
I would achieve that somebody can set an array as `$data` param. Because the type is mixed so this is possible. Now it will throw an exception because the `Tool\Serialize::unserialize($data)` call only accept a string or a null parameter.

But this was it also before your change. So I think we can leave it like that for now.